variables-extensions.tf

```diff
@@ -0,0 +1,179 @@
+# Copyright (c) 2017, 2023 Oracle Corporation and/or its affiliates.
+# Licensed under the Universal Permissive License v 1.0 as shown at https://oss.oracle.com/licenses/upl
+
+variable "check_node_active" {
+  default     = "none"
+  description = "check worker node is active"
+  type        = string
+
+  validation {
+    condition     = contains(["none", "one", "all"], var.check_node_active)
+    error_message = "Accepted values are none, one or all."
+  }
+}
+
+variable "upgrade_nodepool" { # TODO Update
+  default     = false
+  description = "Whether to upgrade the Kubernetes version of the node pools."
+  type        = bool
+}
+
+variable "node_pools_to_drain" { # TODO Update
+  default     = ["none"]
+  description = "List of node pool names to drain during an upgrade. This list is used to determine the worker nodes to drain."
+  type        = list(string)
+}
+
+# Oracle Container Image Registry (OCIR)
+variable "email_address" {
+  default     = "none"
+  description = "The email address used for OCIR."
+  type        = string
+}
+
+variable "secret_id" {
```

Comment:
I think we need to rename this too: maybe `authtoken_id`?

Reply:
More planned for extensions; let's do this separate from this change.